plugins/sh/resources/META-INF/plugin.xml

```diff
@@ -16,8 +16,16 @@ Adds support for working with shell script files
 <li>Dedicated run/debug configuration for shell scripts</li>
 <li>Integration with external tools (<a href="https://github.com/koalaman/shellcheck">ShellCheck</a>, <a href="https://github.com/mvdan/sh">Shfmt</a>, <a href="https://explainshell.com/">Explainshell</a>)
 </li></ul>]]></description>
+
+  <extensionPoints>
+    <extensionPoint name="shSupport" interface="com.intellij.sh.ShSupport" dynamic="true"/>
+  </extensionPoints>
+
+  <extensions defaultExtensionNs="com.jetbrains.sh">
+    <shSupport id="shSupport" implementation="com.intellij.sh.ShSupport$Impl"/>
+  </extensions>
+  
   <extensions defaultExtensionNs="com.intellij">
-    <applicationService serviceInterface="com.intellij.sh.ShSupport" serviceImplementation="com.intellij.sh.ShSupport$Impl"/>
```

Comment:
<sh.shSupport id="shSupport" implementation="com.intellij.sh.ShSupport$Impl" />. This can be here